apps/pwa/src/pages/profile.astro

```diff
@@ -0,0 +1,314 @@
+                                    ></path>
+                                </svg>
+                                <span class="font-medium text-gray-900 text-xs sm:text-sm">Calendar</span>
+                            </a>
+
+                            <button
+                                id="logoutBtn"
+                                class="flex flex-col items-center justify-center p-4 bg-red-50 hover:bg-red-100 rounded-lg transition-colors duration-200 text-center"
+                            >
+                                <svg class="w-6 h-6 text-red-600 mb-2" fill="none" stroke="currentColor" viewBox="0 0 24 24">
+                                    <path
+                                        stroke-linecap="round"
+                                        stroke-linejoin="round"
+                                        stroke-width="2"
+                                        d="M17 16l4-4m0 0l-4-4m4 4H7m6 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h4a3 3 0 013 3v1"
+                                    ></path>
+                                </svg>
+                                <span class="font-medium text-red-900 text-xs sm:text-sm">Logout</span>
+                            </button>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </main>
+</Layout>
+
+<script src="../lib/auth-client.ts"></script>
+<script>
+    // Store the last loaded user ID to detect account changes
+    let lastLoadedUserId: string | null = null;
+
+    async function waitForAuthClient(): Promise<boolean> {
+        return new Promise((resolve) => {
+            if (typeof window !== 'undefined' && window.authClient) {
+                resolve(true);
+                return;
+            }
+            
+            const checkInterval = setInterval(() => {
+                if (typeof window !== 'undefined' && window.authClient) {
+                    clearInterval(checkInterval);
+                    resolve(true);
+                }
+            }, 100);
+            
+            setTimeout(() => {
+                clearInterval(checkInterval);
+                resolve(false);
+            }, 5000);
+        });
+    }
+
+    function getUserIdFromToken(token: string): string | null {
+        try {
+            const tokenParts = token.split('.');
+            if (tokenParts.length === 3) {
+                // JWT token - decode it
+                const payload = tokenParts[1];
+                const base64 = payload.replace(/-/g, '+').replace(/_/g, '/');
+                const padded = base64 + '=='.substring(0, (4 - base64.length % 4) % 4);
+                const decoded = atob(padded);
+                const tokenPayload = JSON.parse(decoded);
+                return tokenPayload.sub || null;
+            }
+        } catch (error) {
+            console.error('Error decoding token:', error);
+        }
+        return null;
+    }
```

Comment:
Can you make sure no functions are repeated across pages? For example, this `getUserIdFromToken` can also be found `my-events`.